### PR TITLE
Update CI to build for all matrix targets

### PR DESCRIPTION
**Pull request by eager-signal**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -156,6 +157,14 @@ jobs:
...
           rust: stable
           os: macos-latest
+        - thing: x86_64-mingw
+          target: x86_64-pc-windows-gnu
+          rust: stable
+          os: windows-latest
+          custom_env:
+            CC: gcc
+            CXX: g++
+            PATH: "C:\msys64\usr\bin;$env:PATH"
```
Comment (jrose-signal) [inline .github/workflows/ci.yml:167]: I think you need to escape this if you put it in quotes. (I also doubt the `$env:PATH` part will work, the env setting happens before GitHub launches the shell.)

Comment (eager-signal) [reply .github/workflows/ci.yml]: Yep and yep. One more attempt, for science.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -211,15 +211,14 @@ jobs:
...
+      run: cargo build --target ${{ matrix.target }} --tests ${{ matrix.extra_test_args }}
       shell: bash
       env: ${{ matrix.custom_env }}
-    - if: matrix.check_only
-      # We `build` because we want the linker to verify we are cross-compiling correctly.
-      run: cargo build --target ${{ matrix.target }} --tests
-      name: Build tests
+    - name: Run tests
+      if: "!matrix.check_only"
+      run: cargo test --target ${{ matrix.target }} ${{ matrix.extra_test_args }}
```
Comment (jrose-signal) [inline .github/workflows/ci.yml:221]: `cargo test` always builds if the tests are stale, which they will be if you change `CXX`.

Comment (eager-signal) [reply .github/workflows/ci.yml:221]: `CXX` is unchanged between the two steps, so `Run tests` is pretty fast (e.g. [this run](https://github.com/signalapp/boring/actions/runs/6866495218/job/18676118339#step:11:19))

Comment (jrose-signal) [reply .github/workflows/ci.yml:221]: Ah, that comment was on an earlier version where you didn't carry over `custom_env`. What you have seems fine.

**Comment by eager-signal** [comment]
This branch was merged upstream

**Pull request closed without merging**